### DISCUSSION TITLE
Firmware upgrade - some issues solved (x1g4 3.7 Hybrid)

### DISCUSSION THREAD

**Discussion by richsimpson** [wills106/homeassistant-solax-modbus#297] [General]
I have posted about a few issues I have had on the firmware's pushed out by Solax. Along with some automations I have created to work around them.

From the original version which prevented the solar receiving more that 3.7 kw of power which was solved with a firmware update that seemed to then break other things, where batteries charged only to 98% instead of 100%, and other points where the inverter lands in idle mode unnecessarily and also the battery not being utilised unless the house demand/load was significant - which had resulted in the power coming fro the grid for hours unnecessarily.

On Friday I got fed up as my automations kept being run to force the battery to kick in, it was a relatively bright day so the battery got back to 100% and then got stuck. My automation had to run to force a high load in the house so it would 'jump start' the battery. I called Solax support and a new firmware was pushed out which (so far) appears to be ok! I haven't tested the solar/PV permitting more than 3.7kw power as no sun. BUT the idle issue, battery failing to kick in and the max charge all appear good. 

For Fri/Sat/Sun evenings, everything has behaved AND my automations and work arounds have not been required. None have been triggered.

Posting the version numbers as may be helpful - or if not at least a memory for me of what changed. I enabled the various sensors in this integration before they started the updates so I could see what changed.

firmware version manager went from 27 to **28.** 
firmware version inverter master from 29 to **30**
firmware version modbus tcp minor is 7 (no change?)
major is 0 (no change?)
solax bootloader version is 2 (no change?)

Then in the solax cloud app the battery was main version from 2.10 to **2.11** and slave version 2.04 to **2.07** on all the batteries ( I have 3 but shows as 6)

**Comment by mlduncombe** [comment]
X1 G4 6.0D definitely seen the issue where a new high load needs to come online to trigger the battery discharge. I've seen #700W load pull from the grid rather than battery but switching on an oil radiator and pulling another 500W will kick the battery into life. Most of the time a much smaller load will kick the battery into life.

Not seen the 98% issue.

inverter master: 24
version manager : 21
battery main: 2.10 (recently updated by Solax after I reported ~25% missing battery capacity, I then manually balanced the batteries which found the missing 25% )
slave v1: 2.02
slave v2: 2.04
